### PR TITLE
Implement linear classifier with SVM loss.

### PR DESCRIPTION
**Pull request by simonb83**

### CODE REVIEW

File: capstone/nn/loss_functions.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    loss = 0.0
+    dW = np.zeros(W.shape)
+    n_samples = X.shape[0]
+    scores = np.dot(X, W)
+    scores -= np.max(scores)
+    exp_scores = np.exp(scores)
+    actual_scores = np.choose(y, scores.T)
+    exp_actual_scores = np.exp(actual_scores)
+    # Sum over all scores; add small number in case 0 for division purposes
+    exp_sum = np.sum(exp_scores, axis=1) + 1e-8
```
Comment (ziky90) [inline capstone/nn/loss_functions.py]: I'd rather go with even smaller number. (eg. `1e-15`)

File: capstone/nn/loss_functions.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    Average loss (float)
+    Gradient matrix with same shape as W
+    """
+    loss = 0.0
+    dW = np.zeros(W.shape)
+    n_samples = X.shape[0]
+    scores = np.dot(X, W)
+    scores -= np.max(scores)
+    exp_scores = np.exp(scores)
+    actual_scores = np.choose(y, scores.T)
```
Comment (ziky90) [inline capstone/nn/loss_functions.py]: I am not sure with this line. I must admit that I'm not that much familiar with `np.choose` and it's a bit difficult for me to imagine what is going on here.

Comment (ziky90) [inline capstone/nn/loss_functions.py:74]: I am not that familiar with `np.choose` so I'm not 100% sure here, but I assume that you're selecting scores only for true labels? 
I'd rather double check this line!

**Comment by ziky90** [comment]
The softmax looks good, but I'm not 100% sure that there is no bug.It'd need either debuguing using prints or using debuger in pycharm.

Otherwise take http://cs231n.github.io/neural-networks-case-study/ as a reference. 

I'd just pay attention to things like division (double check, that output is alway float), also `numpy` operations should be double checked if the output of the loss function looks suspicious.
